lib/filters/substring.js

```diff
@@ -12,87 +12,131 @@ const getAttributeValue = require('../utils/get-attribute-value')
+   * @property {string} [initial] Text that must appear at the start
+   * of a value and may not overlap any value of `any` or `final`.
+   * @property {string} [subInitial] Deprecated, use `initial`.
+   * @property {string[]} [any] Text items that must appear in the
+   * attribute value that do not overlap with `initial`, `final`, or
+   * any other `any` item.
+   * @property {string[]} [subAny] Deprecated, use `any`.
+   * @property {string} [final] Text that must appear at the end of
+   * the attribute value. May not overlap with `initial` or any `any`
    * item.
+   * @property {string} [subFinal] Deprecated, use `final`.
    */
 
   /**
    * @param {SubstringParams} input
    *
    * @throws When any input parameter is of an incorrect type.
    */
-  constructor ({ attribute, subInitial, subAny = [], subFinal } = {}) {
+  constructor ({ attribute, initial, subInitial, any = [], subAny = [], final, subFinal } = {}) {
+    initial = subInitial || initial
+    final = subFinal || final
+
     if (typeof attribute !== 'string' || attribute.length < 1) {
       throw Error('attribute must be a string of at least one character')
     }
+    if (Array.isArray(any) === false) {
+      throw Error('any must be an array of items')
+    }
     if (Array.isArray(subAny) === false) {
       throw Error('subAny must be an array of items')
     }
-    if (subFinal && typeof subFinal !== 'string') {
-      throw Error('subFinal must be a string')
+    if (final && typeof final !== 'string') {
+      throw Error('final must be a string')
     }
 
+    any = subAny.length > 0 ? subAny : any
+
     super({ attribute })
 
-    this.#subInitial = subInitial
-    Array.prototype.push.apply(this.#subAny, subAny)
-    this.#subFinal = subFinal
+    this.#initial = initial
+    Array.prototype.push.apply(this.#any, any)
+    this.#final = final
 
     Object.defineProperties(this, {
       TAG: { value: search.FILTER_SUBSTRINGS },
       type: { value: 'SubstringFilter' }
     })
   }
 
+  /**
+   * @type {string}
+   */
+  get initial () {
+    return this.#initial
+  }
+
+  /**
+   * @type {string[]}
+   */
+  get any () {
+    return this.#any
+  }
+
+  /**
+   * @type {string}
+   */
+  get final () {
+    return this.#final
+  }
+
+  /**
+   * @deprecated use `initial`
+   * @type {string}
+   */
   get subInitial () {
-    return this.#subInitial
+    return this.#initial
   }
 
+  /**
+   * @deprecated use `any`
+   * @type {string[]}
+   */
   get subAny () {
-    return this.#subAny
+    return this.#any
   }
 
+  /**
+   * @deprecated use `final`
+   * @type {string}
+   */
   get subFinal () {
-    return this.#subFinal
+    return this.#final
   }
 
   get json () {
     return {
       type: this.type,
-      subInitial: this.#subInitial,
-      subAny: this.#subAny,
-      subFinal: this.#subFinal
+      initial: this.#initial,
+      any: this.#any,
+      final: this.#final
     }
   }
```

Comment:
Should we still return the prefixed fields here for maximum backwards compatibility?

Reply:
That's a good question that I'm struggling to answer. Since this is an accessor and not a method, we can't change the signature to something like `function json({ useSubPrefix = true } = {})`. But removing the prefixes here is technically a semver major that we are trying to avoid for now. However, no one has noticed, or at least reported, this discrepancy until you. So I'm keen to say that it won't make a difference and we should go ahead and return `{ initial, any, final }`.

Reply:
Are you sure? The discrepancy existed between the code and the (external) documentation. The code itself was coherent and predictable. With this change users could be confused if they construct with the prefix but `.json` omits it. It could also theoretically break existing code, if someone for whatever reason relies on `.json`.

Maybe the most userfriendly way would be to store wether the "sub" prefix was used in the constructor and return the json based off of that? 

```javascript
const f1 = new SubstringFilter({
  attribute: 'foo',
  initial: 'bar',
  any: ['zig', 'zag'],
  final: 'baz'
})

f1.json // {
        //   initial: 'bar',
        //   any:  ['zig', 'zag'],
        //   final: 'baz'
        // }

const f2 = new SubstringFilter({
  attribute: 'foo',
  subInitial: 'bar',
  subAny: ['zig', 'zag'],
  subFinal: 'baz'
})

f2.json // {
        //   subInitial: 'bar',
        //   subAny:  ['zig', 'zag'],
        //   subFinal: 'baz'
        // }
```

This would prevent existing code from breaking, wouldn't it? Or am I overthinking this and nobody will care? 😃

Reply:
Sounds like a reasonable compromise to me. Good idea 🎉

Reply:
I updated the PR accordingly. Let me know what you think 🙂